beacon_node/beacon_chain/test_harness/src/validator/validator.rs

```diff
@@ -0,0 +1,133 @@
+use super::direct_beacon_node::DirectBeaconNode;
+use super::direct_duties::DirectDuties;
+use super::signer::TestSigner;
+use attester::PollOutcome as AttestationPollOutcome;
+use attester::{Attester, Error as AttestationPollError};
+use beacon_chain::BeaconChain;
+use block_producer::PollOutcome as BlockPollOutcome;
+use block_producer::{BlockProducer, Error as BlockPollError};
+use db::MemoryDB;
+use slot_clock::TestingSlotClock;
+use std::sync::Arc;
+use types::{BeaconBlock, ChainSpec, FreeAttestation, Keypair};
+
+#[derive(Debug, PartialEq)]
+pub enum BlockProduceError {
+    DidNotProduce(BlockPollOutcome),
+    PollError(BlockPollError),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum AttestationProduceError {
+    DidNotProduce(AttestationPollOutcome),
+    PollError(AttestationPollError),
+}
+
+/// A `BlockProducer` and `Attester` which sign using a common keypair.
+///
+/// The test validator connects directly to a borrowed `BeaconChain` struct. It is useful for
+/// testing that the core proposer and attester logic is functioning. Also for supporting beacon
+/// chain tests.
+pub struct TestValidator {
```

Comment:
rename to `validator` or file to `test_validator`?

Reply:
`Validator` clashes with `types::Validator` and `test_validator.rs` looks like a test. I'll try figure something out, lemme know if you have any ideas.

Reply:
`validator_test` ?